File: spring-boot-app/src/main/java/org/example/angulardemo/entity/Product.java
package org.example.angulardemo.entity;

import jakarta.persistence.Entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@Entity
@ToString
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;
    @NotEmpty
    @Column(nullable = false)
    private String extId;
    @NotEmpty
    @Column(nullable = false)
    private String name;
    private String description;

}

File: spring-boot-app/src/main/kotlin/org/example/angulardemo/entity/Product.kt
package org.example.angulardemo.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table


@Table(name = "product")
data class Product(
    @Id
    val id: String?,
    val userId: Long,
    var externalId: Long?,
    var name: String,
    var description: String?,
)
